cmd/uid_provider/client/client.go

```diff
@@ -0,0 +1,306 @@
+package client
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/go-redis/redis/v8"
+	"gitlab.inspr.dev/inspr/core/pkg/api/auth"
+	"gitlab.inspr.dev/inspr/core/pkg/controller/client"
+	"gitlab.inspr.dev/inspr/core/pkg/rest/request"
+)
+
+// Client defines a Redis client, which has the interface methods
+type Client struct {
+	rdb *redis.Client
+}
+
+// NewRedisClient creates and returns a new Redis client
+func NewRedisClient() *Client {
+	host := os.Getenv("REDIS_HOST")
+	port := os.Getenv("REDIS_PORT")
+	password := os.Getenv("REDIS_PASSWORD")
+
+	return &Client{
+		rdb: redis.NewClient(&redis.Options{
+			Addr:     host + ":" + port,
+			Password: password,
+			DB:       0, // use default DB
+		}),
+	}
+}
+
+// CreateUser inserts a new user into Redis
+func (c *Client) CreateUser(ctx context.Context, uid, pwd string, newUser User) error {
+	usrAuthorized, err := havePermission(ctx, c.rdb, uid, pwd)
+
+	if usrAuthorized {
+		if err := set(ctx, c.rdb, newUser); err != nil {
+			return err
+		}
+		return nil
+	}
+	return err
+}
+
+// DeleteUser deletes an user from Redis, if it exists
+func (c *Client) DeleteUser(ctx context.Context, uid, pwd, usrToBeDeleted string) error {
+	usrAuthorized, err := havePermission(ctx, c.rdb, uid, pwd)
+
+	if usrAuthorized {
+		if err := delete(ctx, c.rdb, usrToBeDeleted); err != nil {
+			return err
+		}
+		return nil
+	}
+	return err
+}
+
+// UpdatePassword changes an users password, if that user exists
+func (c *Client) UpdatePassword(ctx context.Context, uid, pwd, usrToBeUpdated, newPwd string) error {
+	usrAuthorized, err := havePermission(ctx, c.rdb, uid, pwd)
```

Comment:
I prefer naming it like, `isAuthorized`, but that's just a thought

Reply:
I agree :smile: